### PR TITLE
fix: migration issues

### PR DESCRIPTION
**Pull request by christyjacob4**
## What does this PR do?

When running migrations more than once in the same major version, document values were being overwritten

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [ ] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -4,23 +4,21 @@
...
-* Fix create phone session abuse key by @stnguyen90 in https://github.com/appwrite/appwrite/pull/6134
-* Fix CLI backwards compatibility by @stnguyen90 in https://github.com/appwrite/appwrite/pull/6125
-* Override forEachDocument() to skip the cache collection by @stnguyen90 in https://github.com/appwrite/appwrite/pull/6144
-* Fix Not Found error when deploying function from git by @stnguyen90 in https://github.com/appwrite/appwrite/pull/6133
-* Add required params for scheduled functions by @stnguyen90 in https://github.com/appwrite/appwrite/pull/6148
-* Update the error message for router_domain_not_configured by @stnguyen90 in https://github.com/appwrite/appwrite/pull/6145
-* Fix _APP_EXECUTOR_HOST for upgrades by @stnguyen90 in https://github.com/appwrite/appwrite/pull/6141
-* Change executor hostname back to appwrite-executor by @stnguyen90 in https://github.com/appwrite/appwrite/pull/6160
-* Fix create execution request filter from previous SDK version by @stnguyen90 in https://github.com/appwrite/appwrite/pull/6146
-* Fix: AI Assistant by @Meldiron in https://github.com/appwrite/appwrite/pull/6153
```
Comment (stnguyen90) [inline CHANGES.md:16]: I think we're missing the AI Assistant fix

File: CHANGES.md [hunk 2]
```diff
@@ -4,23 +4,21 @@
...
+* Update the error message for router_domain_not_configured [#6145](https://github.com/appwrite/appwrite/pull/6145)
+* Fix _APP_EXECUTOR_HOST for upgrades [#6141](https://github.com/appwrite/appwrite/pull/6141)
+* Change executor hostname back to appwrite-executor [#6160](https://github.com/appwrite/appwrite/pull/6160)
+* Fix create execution request filter from previous SDK version [#6146](https://github.com/appwrite/appwrite/pull/6146)
+* Make URL optional for Create Membership API and Serverside Requests [#6157](https://github.com/appwrite/appwrite/pull/6157)
+* Support for v2 functions [#6142](https://github.com/appwrite/appwrite/pull/6142)
+* Fix migrations worker [#6116](https://github.com/appwrite/appwrite/pull/6116)
+* Fix: Global variables by [#6150](https://github.com/appwrite/appwrite/pull/6150)
+* Fix webhook secret validation and executor path validation [#6162](https://github.com/appwrite/appwrite/pull/6162)
+* Fix: Untrusted custom domains + auto-ssl [#6155](https://github.com/appwrite/appwrite/pull/6155)
```
Comment (stnguyen90) [inline CHANGES.md:21]: The fixes should be moved to the Fixes section, right?

File: src/Appwrite/Migration/Version/V19.php [hunk 3]
```diff
@@ -651,7 +651,7 @@ protected function fixDocument(Document $document): Document
         switch ($document->getCollection()) {
             case 'attributes':
             case 'indexes':
-                $status = $document->getAttribute('status', '');
+                $status = $document->getAttribute('status', $document->getAttribute('status', ''));
                 if ($status === 'failed') {
                     $document->setAttribute('error', 'Unknown problem');
                 }
```
Comment (stnguyen90) [inline src/Appwrite/Migration/Version/V19.php:657]: This isn't required, but checking the default for the error attribute is:

```suggestion
                $status = $document->getAttribute('status');
                if ($status === 'failed') {
                    $document->setAttribute('error', $document->getAttribute('error', 'Unknown problem'));
                }
```

File: src/Appwrite/Migration/Version/V19.php [hunk 4]
```diff
@@ -706,7 +705,7 @@ protected function fixDocument(Document $document): Document
...
 
                 $commands = $this->getFunctionCommands($document);
-                $document->setAttribute('commands', $commands);
+                $document->setAttribute('commands', $document->getAttribute('commands', $commands));
```
Comment (stnguyen90) [inline src/Appwrite/Migration/Version/V19.php:708]: We need to avoid creating a schedule if a schedule already exists 👇🏼

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

File: CHANGES.md (hunk 2 above)
Comment (christyjacob4) [reply CHANGES.md:21]: moved 👍

File: CHANGES.md (hunk 1 above)
Comment (christyjacob4) [reply CHANGES.md:16]: added 👍

File: src/Appwrite/Migration/Version/V19.php (hunk 4 above)
Comment (christyjacob4) [reply src/Appwrite/Migration/Version/V19.php:708]: added a check for it

**Review by stnguyen90 — APPROVED** [review]

File: src/Appwrite/Migration/Version/V19.php [hunk 5]
```diff
@@ -706,35 +705,38 @@ protected function fixDocument(Document $document): Document
...
-                    'projectId' => $this->project->getId(),
-                    'schedule'  => $document->getAttribute('schedule'),
-                    'active' => !empty($document->getAttribute('schedule')) && !empty($document->getAttribute('deployment')),
-                ]));
-
-                $document->setAttribute('scheduleId', $schedule->getId());
-                $document->setAttribute('scheduleInternalId', $schedule->getInternalId());
+                $document->setAttribute('commands', $document->getAttribute('commands', $commands));
+
+                if ($document->getAttribute('schedule') && !$document->getAttribute('scheduleId', null)) {
```
Comment (stnguyen90) [inline src/Appwrite/Migration/Version/V19.php:710]: schedule doesn't matter when determining whether a schedule document should exist:

```suggestion
                if (empty($document->getAttribute('scheduleId', null))) {
```

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

**Pull request merged**